hud/weapon_hotbar.gd

```diff
@@ -29,3 +47,11 @@ func _get_items() -> Array[WeaponHotbarItem]:
 		if child is WeaponHotbarItem:
 			items.append(child)
 	return items
+
+
+## Can return null!
+func _get_player() -> Player:
+	var nodes = self.get_tree().get_nodes_in_group("player")
+	if nodes.size() == 0:
+		return null
+	return nodes[0] as Player
```

Comment:
is this so we don't have to have a direct reference like `%Player`? should we instead ensure the `nodes.size` is `== 1`? or maybe log a warning in case we ever add anything else to that group before this `_get_player` is called and could potentially not get something that doesn't necessarily extend/implement `Player`?

Reply:
Yeah, the thing with unique names was a bit tricky, because we don't create the player from the editor itself, so I figured this was the easiest way to get that, otherwise it would have to be a `get_node(absolute_path)` and in that case if the path changes all references get broken, so not sure.

The other thing is that, really this is needs the `self.get_tree` call, but otherwise this method is completely agnostic to this particular node and could be a util instead, and in that case, having a cache reference node or a warning if there are multiples makes absolutely a ton of sense.

Not sure at this point what to do, so i'll leave it as is it for now.